sql_joins/sql_joins.md

```diff
@@ -0,0 +1,875 @@
+
+Your criteria might be something like:
+
+* Join the rows from "depression\_scale" with rows from "subject\_address" **... but only if the subject ID field matches**
+* Join the rows from "items" with the rows from "orders" **... but only if the item ID field matches**
+* Join the rows from "biology\_grades" with the rows from "psychology\_grades" **... but only if the student ID is the same and the semester is the same**
+
+When the conditions in your join criteria evaluate as TRUE for a row then a join will be performed for those rows, and when the join criteria are evaluated as FALSE no join for those rows will take place.
+
+**Data Relationships**
+
+As a reminder, SQL is a **relational database**, so it's not surprising that we talk about data relationships in this module.  Equality is one kind of relationship, when two data points are identical, but other relationships, like "less than" or "between" will also prove useful when we set up our join criteria.
+
+Join criteria will be some sort of relationship statement referencing data that occurs in both tables you want to join.  This relationship statment will be valuated to TRUE or FALSE when  your join is executed.  Often, the relationship is equality -- you're looking for a perfect match.  We'll start with equality, the most frequently used condition, on the next page. 
+
+#### Equality, Example 1
+
+Do you have subject identifiers or student ID numbers in two different tables?  This shared information can be used to connect (join) data from these tables, based on the identifier being equal.  
+
+For example, if the subject ID matches, a row from table A and a row from table B will be joined.  If the subject ID doesn't match, these rows won't be joined.  Maybe we're trying to match lung cancer occurrence and smoking exposure in the same row:
+
+**disease**
+
+<!-- data-type="none" class="tight-table"-->
+| subject\_id  | lung\_cancer |
+| :--------- | :--------- | 
+| 3  | TRUE |
+| 5 | FALSE  |
+| 8  | FALSE |
+
+**smoking**
+
+<!-- data-type="none" class="tight-table"-->
+| subject\_id  | smoking\_pack\_years |
+| :--------- | :--------- | 
+| 2  | 10 |
+| 3 | 10  |
+| 4  | 0 |
+
+
+In code, we usually use an **ON** statement:
+
+```sql
+...
+ON disease.subject_id = smoking.subject_id
+```
+
+In this case, we're comparing the equality of two fields that **have the same name**, so we could also use **USING**.  This special word only applies when you're looking for a perfect match between fields that have matching names, too.  It's okay if you never use `USING` and prefer to always stick with `ON`, which is more multi-purpose.
+
+```sql
+...
+USING(subject_id)
+```
+
+With either of the above code snippets:
+
+* subject 3 from disease matches with subject 3 from smoking <i class="fa fa-check" aria-hidden="true"></i>
```

Comment:
Does the `aria-hidden="true"` tell screenreader what to say? That is cool!

Reply:
No, I think it means "hide this graphic" but honestly I just copy-pasted from the FA website :)

Reply:
in that case, I think all of these fun check/X marks should be accompanied or replaced by a word like "included" or "excluded" because the symbol is the only indication of which are included/excluded. 

See this conversation about how these need alt text: https://stackoverflow.com/questions/29393575/font-awesome-support-for-screen-readers-and-accessibility

Reply:
I think the best thing is just to delete... these are really decorations that don't add much, really.